### PR TITLE
Add API versioning support

### PR DESCRIPTION
**Pull request by ndhoule**
- Fixes fixtures to not be dynamically modified
- Tests both legacy and modern API formats
- General maintenance (remove unused modules, refactor tests to look like other integrations, etc.)

### CODE REVIEW

**Comment by ndhoule** [comment]
cc @calvinfo @gjohnson

File: lib/index.js [hunk 1]
```diff
@@ -15,6 +14,7 @@ var ms = require('ms');
 
 var Webhooks = module.exports = integration('Webhooks')
   .channels(['server', 'mobile', 'client'])
+  .ensure('settings.version')
```
Comment (gjohnson) [inline lib/index.js]: This will bubble back an error if it's missing, did we add it to all the mongo documents?

Comment (ndhoule) [inline lib/index.js]: not yet, I have a Trello item for that before I deploy, though

https://trello.com/c/xSg6pBq3/350-webhooks-legacy-transform-removal

File: lib/index.js [hunk 2]
```diff
@@ -41,33 +41,33 @@ Webhooks.prototype.screen = request;
...
 
 function request(message, fn){
-  var json = message.json();
-  var data;
+  var data = message.json();
   var err;
 
-  json.options = json.options || json.context;
+  // TODO: Remove this once v1 support is officially deprecated
+  if (this.settings.version === 1) {
```
Comment (gjohnson) [inline lib/index.js]: I probably wouldn't trust this to be strictly a number coming from mongo (flexible schemas are good right? haha) Probably want to do it loosely or `parseInt(this.settings.version, 10)`

Comment (ndhoule) [inline lib/index.js]: Metadata is going to look like this:

```json
    "version": {
      "default": 2,
      "description": "The Segment API version, which determines the format we'll send your data in. Select `v1` for our legacy format, and `v2` for our modern, [spec](https://segment.com/docs/spec/)-compliant format. *Note*: This option exists for legacy purposes only and will be removed in the future. Switch from `v2` at your own peril!",
      "label": "Segment API Version",
      "type": "select",
      "options": [
        {
          "text": "v1",
          "value": 1
        },
        {
          "text": "v2",
          "value": 2
        }
      ]
    }
```

So it should be a non-issue? Still doesn't hurt to add it, haha

Comment (calvinfo) [inline lib/index.js]: nice, sgtm! I'd also say we should start switching these to be strings, and set everyone to "1"

Comment (ndhoule) [inline lib/index.js]: makes sense to me

**Comment by ndhoule** [comment]
thanks y'all!

**Pull request merged**